utils/split_c_s_to_ranges.py

```diff
@@ -0,0 +1,34 @@
+# pylint: skip-file
+import json
+
+loaders = 4
+multiplier = 1
+count = loaders*multiplier
+num_of_rows = 1300000000
+gauss_ratio = 0.015
```

Comment:
1.5% from what I spoke with @roydahan , but I could misunderstood. Anyway this will need to be adapted to reach lower cache hit ratio (if we aim for 50%)

Reply:
I think the math is little off here,

I've tweaked it a bit by following the docs in https://cassandra.apache.org/doc/stable/cassandra/tools/cassandra_stress.html:
```diff
diff --git a/utils/split_c_s_to_ranges.py b/utils/split_c_s_to_ranges.py
index 56703c75b..d29aab59a 100644
--- a/utils/split_c_s_to_ranges.py
+++ b/utils/split_c_s_to_ranges.py
@@ -1,11 +1,11 @@
 # pylint: skip-file
 import json

-loaders = 4
+loaders = 6
 multiplier = 1
 count = loaders*multiplier
-num_of_rows = 1300000000
-gauss_ratio = 0.015
+num_of_rows = 650000000
+gauss_ratio = 33.3333332821
 rows_per_command = num_of_rows/count
 write_cs = "cassandra-stress write no-warmup cl=QUORUM duration=60m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=340 -col 'size=FIXED(128) n=FIXED(8)'"
 read_cs = "cassandra-stress read no-warmup cl=QUORUM duration=60m -mode cql3 native -rate threads=350 -col 'size=FIXED(128) n=FIXED(8)'"
@@ -19,16 +19,20 @@ print(json.dumps(commands, indent=2))

 commands = []
 for idx in range(count):
-    rows = idx*rows_per_command
+    min_row = idx * rows_per_command + 1
+    max_row = idx * rows_per_command + rows_per_command
+    mean = (min_row + max_row) / 2
     commands.append(
-        f"{read_cs} -pop 'dist=gauss({rows + 1:.0f}..{rows + rows_per_command:.0f},{rows + rows_per_command / 2:.0f},{rows + rows_per_command * gauss_ratio:.0f})'")
+        f"{read_cs} -pop 'dist=gauss({min_row:.0f}..{max_row:.0f},{mean:.0f},{(mean - min_row) / gauss_ratio:.0f})'")
 print("Read:")
 print(json.dumps(commands, indent=2))

 commands = []
 for idx in range(count):
-    rows = idx*rows_per_command
+    min_row = idx * rows_per_command + 1
+    max_row = idx * rows_per_command + rows_per_command
+    mean = (min_row + max_row) / 2
     commands.append(
-        f"{mixed_cs} -pop 'dist=gauss({rows + 1:.0f}..{rows + rows_per_command:.0f},{rows + rows_per_command / 2:.0f},{rows + rows_per_command * gauss_ratio:.0f})'")
+        f"{mixed_cs} -pop 'dist=gauss({min_row:.0f}..{max_row:.0f},{mean:.0f},{(mean - min_row) / gauss_ratio:.0f})'")
 print("Mixed:")
 print(json.dumps(commands, indent=2))
```

and now the results is:

```
Read:
[
  "cassandra-stress read no-warmup cl=QUORUM duration=60m -mode cql3 native -rate threads=350 -col 'size=FIXED(128) n=FIXED(8)' -pop 'dist=gauss(1..108333333,54166667,1625000)'",
  "cassandra-stress read no-warmup cl=QUORUM duration=60m -mode cql3 native -rate threads=350 -col 'size=FIXED(128) n=FIXED(8)' -pop 'dist=gauss(108333334..216666667,162500000,1625000)'",
  "cassandra-stress read no-warmup cl=QUORUM duration=60m -mode cql3 native -rate threads=350 -col 'size=FIXED(128) n=FIXED(8)' -pop 'dist=gauss(216666668..325000000,270833334,1625000)'",
  "cassandra-stress read no-warmup cl=QUORUM duration=60m -mode cql3 native -rate threads=350 -col 'size=FIXED(128) n=FIXED(8)' -pop 'dist=gauss(325000001..433333333,379166667,1625000)'",
  "cassandra-stress read no-warmup cl=QUORUM duration=60m -mode cql3 native -rate threads=350 -col 'size=FIXED(128) n=FIXED(8)' -pop 'dist=gauss(433333334..541666667,487500000,1625000)'",
  "cassandra-stress read no-warmup cl=QUORUM duration=60m -mode cql3 native -rate threads=350 -col 'size=FIXED(128) n=FIXED(8)' -pop 'dist=gauss(541666668..650000000,595833334,1625000)'"
]
```

I think it might explain the cache hit ratio, I think the `stdev` should be identical figure across all of the commands

Reply:
fixed